common/build/build.go

```diff
@@ -25,48 +25,56 @@
 package build
 
 import (
-	"embed"
-	"encoding/json"
+	"runtime/debug"
 	"time"
-
-	"go.temporal.io/server/build/cgo"
-)
-
-const (
-	buildInfoDataFile = "info/data.json"
 )
 
 type (
 	Info struct {
-		// GitRevision is the git revision associated with this build.
-		GitRevision string
-		// BuildTimeUnix is the seconds since epoch representing the date this build was created.
-		BuildTimeUnix int64
+		Available bool
+		GoVersion string
+		GoArch    string
+		GoOs      string
 		// CgoEnabled indicates whether cgo was enabled when this build was created.
 		CgoEnabled bool
+
+		// GitRevision is the git revision associated with this build.
+		GitRevision string
+		// GitTime is the git revision time.
+		GitTime     time.Time
+		GitModified bool
 	}
 )
 
 var (
-	//go:embed info
-	buildInfoFs embed.FS
-	InfoData    Info
+	InfoData Info
 )
 
 func init() {
-	InfoData = Info{
-		GitRevision:   "unknown",
-		BuildTimeUnix: 0,
-		CgoEnabled:    cgo.Enabled,
-	}
-
-	buildInfoDataJson, err := buildInfoFs.ReadFile(buildInfoDataFile)
-	if err != nil {
+	buildInfo, ok := debug.ReadBuildInfo()
+	if !ok {
 		return
 	}
-	_ = json.Unmarshal(buildInfoDataJson, &InfoData)
-}
 
-func (i Info) BuildTime() time.Time {
-	return time.Unix(i.BuildTimeUnix, 0)
+	InfoData.Available = true
+	InfoData.GoVersion = buildInfo.GoVersion
+
+	for _, setting := range buildInfo.Settings {
+		switch setting.Key {
+		case "GOARCH":
+			InfoData.GoArch = setting.Value
+		case "GOOS":
+			InfoData.GoOs = setting.Value
+		case "CGO_ENABLED":
+			InfoData.CgoEnabled = setting.Value == "1"
+		case "vcs.revision":
+			InfoData.GitRevision = setting.Value
+		case "vcs.time":
+			if gitTime, err := time.Parse(time.RFC3339, setting.Value); err == nil {
+				InfoData.GitTime = gitTime
+			}
+		case "vcs.modified":
```

Comment:
how do you find those keys? any documentation about what are the available keys?

Reply:
No, I wasn't able to find the doc. Release notes just describes it in general and I had to dump entire `settings` struct to see what is there.

Reply:
Also `go version -m temporal-server` shows it.